### ISSUE TITLE
Bug of FashionMNIST

### ISSUE THREAD

**Issue by leftthomas** [pytorch/vision#445]
I have installed PyTorch 0.3.1, torchvision 0.2.0, torchnet 0.0.1, visdom 0.1.7, CUDA9.0, CUDNN 7.0. I wrote some code by using torchnet to train my model, but I found the `FashionMNIST` val images are wrong, and the GPU memory leakage problem, I can't figure out why, I think it maybe a bug of `FashionMNIST` or `DataLoader`.
1.  Run train from terminal, from this screenshot you could see the fashion mnist data set have been downloaded.
<img width="590" alt="qq20180311-093233 2x" src="https://user-images.githubusercontent.com/9991443/37248591-1e42aaca-2511-11e8-9132-4fbcbf2f5a28.png">
2. The used GPU memory at the first epoch,  from this screenshot you could see the used GPU memory is  **755MB**.
<img width="582" alt="qq20180311-093225 2x" src="https://user-images.githubusercontent.com/9991443/37248606-a2ea5e8a-2511-11e8-9ed0-f57109afee4d.png">
3. After one epoch, the used GPU memory increased,  from this screenshot you could see the used GPU memory is  **1913MB**! And I used visdom to see the val images, but you can found that the val images are MNIST not FashionMNIST!
<img width="584" alt="qq20180311-093303 2x" src="https://user-images.githubusercontent.com/9991443/37248769-b8f71ed6-2514-11e8-887e-21bebb10ae7e.png">
<img width="1279" alt="qq20180311-093353 2x" src="https://user-images.githubusercontent.com/9991443/37248770-bfb27e1e-2514-11e8-84ac-dc9ab6ff7433.png">

I have tested it on Ubuntu16.04 single GPU or Multi GPUs and macOS, only Ubuntu have this problem, the macOS haven't this problem.
the reproductive code are here:
[bugs.zip](https://github.com/pytorch/vision/files/1799976/bugs.zip)

**Comment by leftthomas** [comment]
@fmassa , I have reinstalled my Ubuntu16.04 system and reinstall the PyTorch and tested it, the bug is still there! I have checked the md5sum of downloaded FashionMNIST dataset by using the FashionMNIST class, and I found the md5sum is different with the [office FashionMNIST dataset](https://github.com/zalandoresearch/fashion-mnist) !

| Name  | MD5 Checksum | Office MD5 Checksum |
| --- | --- |--- | 
|`train-images-idx3-ubyte.gz`|`f4a8712d7a061bf5bd6d2ca38dc4d50a`|`8d4fb7e6c68d591d4c3dfef9ec88bf0d`|
|`train-labels-idx1-ubyte.gz`|`9018921c3c673c538a1fc5bad174d6f9`|`25c81989df183df01b3e8a0aad5dffbe`|
|`t10k-images-idx3-ubyte.gz`|`2646ac647ad5339dbf082846283269ea`|`bef4ecab320f06d8554ea6380940ec79`|
|`t10k-labels-idx1-ubyte.gz`|`15d484375f8d13e6eb1aabb0c3f46965`|`bb300cfdad3c16e7a12a480ee83cd310`|

Further more, the md5sum value of `t10k-images-idx3-ubyte.gz` is same to MNIST!

**Comment by leftthomas** [comment]
@fmassa , I have figure out the dataset issues, it is solved on [here](https://github.com/zalandoresearch/fashion-mnist/issues/107), but the memory leaky problem still there, I have tried on CUDA9.0, CUDA9.1.

**Comment by fmassa** [comment]
Great that you have found the solution to the downloading problem!
About the memory leak that you mentioned, does it continuously increase over time or is it just during the first evaluation part that the memory gets bumped? If that's the case, it's probably because the reference of your loss during your last iteration during training is still kept alive when you start testing your network, requiring to almost double the amount of memory needed.
That's the first place I'd look at.

**Comment by leftthomas** [comment]
@fmassa , yes, it just during the first evaluation part that the memory gets bumped, I also tested the example of `torchnet`, but I can't find where is different between my code and the example, I tried comment my test apart, and the problem is gone.

**Comment by fmassa** [comment]
you probably need to remove all references to `output`/`loss` etc from your state before performing evaluation.
This is a torchnet specific question though, and I'm not super familiar with it so it might be better asking there.

**Comment by leftthomas** [comment]
@fmassa , I have found the reason, it is a bug of nn.conv2d(), I have tested if the model use nn.conv2d(512,512, ...) then it will encounter the memory problem.

**Comment by fmassa** [comment]
This is probably a bug in either how you are using `tnt` or in `tnt` itself, so I'm closing the issue as it is not related to `torchvision`.
